pkg/controller/hibernation/ibmcloud_actuator.go

```diff
@@ -0,0 +1,173 @@
+package hibernation
+
+import (
+	"context"
+
+	"github.com/IBM/vpc-go-sdk/vpcv1"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	controllerutils "github.com/openshift/hive/pkg/controller/utils"
+	ibmclient "github.com/openshift/hive/pkg/ibmclient"
+)
+
+var (
+	// States described in IBM Cloud API docs
+	// https://cloud.ibm.com/apidocs/vpc?code=go#get-instance
+	ibmRunningStates           = sets.NewString("running")
+	ibmStoppedStates           = sets.NewString("stopped")
+	ibmPendingStates           = sets.NewString("pending")
+	ibmStoppingStates          = sets.NewString("stopping")
+	ibmRunningOrPendingStates  = ibmRunningStates.Union(ibmPendingStates)
+	ibmStoppedOrStoppingStates = ibmStoppedStates.Union(ibmStoppingStates)
+	ibmNotRunningStates        = ibmStoppedOrStoppingStates.Union(ibmPendingStates)
+	ibmNotStoppedStates        = ibmRunningOrPendingStates.Union(ibmStoppingStates)
+)
+
+func init() {
+	RegisterActuator(&ibmCloudActuator{ibmCloudClientFn: getIBMCloudClient})
+}
+
+type ibmCloudActuator struct {
+	// ibmCloudClientFn is the function to build an IBM Cloud client, here for testing
+	ibmCloudClientFn func(*hivev1.ClusterDeployment, client.Client, log.FieldLogger) (ibmclient.API, error)
+}
+
+// CanHandle returns true if the actuator can handle a particular ClusterDeployment
+func (a *ibmCloudActuator) CanHandle(cd *hivev1.ClusterDeployment) bool {
+	return cd.Spec.Platform.IBMCloud != nil
+}
+
+// StopMachines will stop machines belonging to the given ClusterDeployment
+func (a *ibmCloudActuator) StopMachines(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) error {
+	logger = logger.WithField("cloud", "ibmcloud")
+	ibmCloudClient, err := a.ibmCloudClientFn(cd, hiveClient, logger)
+	if err != nil {
+		return err
+	}
+
+	instances, err := getIBMCloudClusterInstances(cd, ibmCloudClient, runningOrPendingStates, logger)
+	if err != nil {
+		return err
+	}
+	if len(instances) == 0 {
+		logger.Warning("No instances were found to stop")
+		return nil
+	}
+	err = ibmCloudClient.StopInstances(instances)
+	if err != nil {
+		logger.WithError(err).Log(controllerutils.LogLevel(err), "failed to stop IBM Cloud instances")
+		return err
+	}
+
+	return nil
+}
+
+// StartMachines will select machines belonging to the given ClusterDeployment
+func (a *ibmCloudActuator) StartMachines(cd *hivev1.ClusterDeployment, hiveClient client.Client, logger log.FieldLogger) error {
+	logger = logger.WithField("cloud", "ibmcloud")
+	ibmCloudClient, err := a.ibmCloudClientFn(cd, hiveClient, logger)
+	if err != nil {
+		return err
+	}
+
+	instances, err := getIBMCloudClusterInstances(cd, ibmCloudClient, stoppedOrStoppingStates, logger)
+	if err != nil {
+		return err
+	}
+	if len(instances) == 0 {
+		logger.Info("No instances were found to start")
```

Comment:
Why is this info but L59 is warning?

Reply:
Good question, this actuator is a monster born of the aws and gcp actuators.

Reply:
I made both of them `Info`.